File: BraintreeCore/src/test/java/com/braintreepayments/api/PreferredPaymentMethodsClientResultUnitTest.java
package com.braintreepayments.api;

import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class PreferredPaymentMethodsClientResultUnitTest {

    @Test
    public void fromJson_whenApiDetectsPayPalPreferred_setsPayPalPreferredToTrue() {
        PreferredPaymentMethodsResult result = PreferredPaymentMethodsResult.fromJSON("{\n" +
                "  \"data\": {\n" +
                "    \"preferredPaymentMethods\": {\n" +
                "      \"paypalPreferred\": true\n" +
                "    }\n" +
                "  }\n" +
                "}", false);

        assertTrue(result.isPayPalPreferred());
    }

    @Test
    public void fromJson_whenApiDetectsPayPalNotPreferred_setsPayPalPreferredToFalse() {
        PreferredPaymentMethodsResult result = PreferredPaymentMethodsResult.fromJSON("{\n" +
                "  \"data\": {\n" +
                "    \"preferredPaymentMethods\": {\n" +
                "      \"paypalPreferred\": false\n" +
                "    }\n" +
                "  }\n" +
                "}", false);

        assertFalse(result.isPayPalPreferred());
    }

    @Test
    public void fromJson_whenVenmoAppIsInstalled_setsVenmoPreferredToTrue() {
        PreferredPaymentMethodsResult result = PreferredPaymentMethodsResult.fromJSON("json", true);
        assertTrue(result.isVenmoPreferred());
    }

    @Test
    public void fromJson_whenVenmoAppIsNotInstalled_setsVenmoPreferredToFalse() {
        PreferredPaymentMethodsResult result = PreferredPaymentMethodsResult.fromJSON("json", false);
        assertFalse(result.isVenmoPreferred());
    }

    @Test
    public void fromJson_whenJsonIsInvalid_setsIsPayPalPreferredToFalse() {
        PreferredPaymentMethodsResult result = PreferredPaymentMethodsResult.fromJSON("invalid-response", false);

        assertFalse(result.isPayPalPreferred());
        assertFalse(result.isVenmoPreferred());
    }
}

File: BraintreeCore/src/test/java/com/braintreepayments/api/PreferredPaymentMethodsClientResultUnitTest.kt
package com.braintreepayments.api

import com.braintreepayments.api.PreferredPaymentMethodsResult.Companion.fromJSON

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class PreferredPaymentMethodsClientResultUnitTest {
    @Test
    fun fromJson_whenApiDetectsPayPalPreferred_setsPayPalPreferredToTrue() {
        val json =
            //language=json
            """{
                  "data": {
                    "preferredPaymentMethods": {
                      "paypalPreferred": true
                    }
                  }
            }"""
            //
        val result = fromJSON(json, false)
        assertTrue(result.isPayPalPreferred())
    }

    @Test
    fun fromJson_whenApiDetectsPayPalNotPreferred_setsPayPalPreferredToFalse() {
        val json =
            //language=json
            """{
                  "data": {
                    "preferredPaymentMethods": {
                      "paypalPreferred": false
                    }
                  }
            }"""
            //
        val result = fromJSON(json, false)
        assertFalse(result.isPayPalPreferred())
    }

    @Test
    fun fromJson_whenVenmoAppIsInstalled_setsVenmoPreferredToTrue() {
        val result = fromJSON("json", true)
        assertTrue(result.isVenmoPreferred())
    }

    @Test
    fun fromJson_whenVenmoAppIsNotInstalled_setsVenmoPreferredToFalse() {
        val result = fromJSON("json", false)
        assertFalse(result.isVenmoPreferred())
    }

    @Test
    fun fromJson_whenJsonIsInvalid_setsIsPayPalPreferredToFalse() {
        val result = fromJSON("invalid-response", false)
        assertFalse(result.isPayPalPreferred())
        assertFalse(result.isVenmoPreferred())
    }
}